src/gebt_poc/quadrature.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "src/gebt_poc/element.h"
+
+namespace openturbine::gebt_poc {
+
+/// An abstract class for providing common interface to numerical quadrature rules
+class Quadrature {
+public:
+    virtual ~Quadrature() = default;
+
+    /// Returns the number of quadrature points
+    virtual size_t GetNumberOfQuadraturePoints() const = 0;
+
+    /// Returns the quadrature points (read only)
+    virtual const std::vector<double>& GetQuadraturePoints() const = 0;
```

Comment:
Should the points and weights be in a view so they can be accessed during integration?

Reply:
Let's add this as a TODO for future, I don't think it's within the scope for this PR.